tools/translation/repair.py

```
@@ -0,0 +1,95 @@
+        assign("translation_status", "original", "set_source_status_original")
+        assign("translation_source_lang", source_lang, "set_source_language")
+    else:
+        assign("translation_source_lang", source_lang, "set_translation_source_language")
+        if source_page:
+            assign("translation_source", source_page.rel_path, "set_translation_source")
+        if not read_scalar(frontmatter, "translation_status"):
+            assign("translation_status", "needs-review", "set_missing_translation_status")
+
+        for key in (BODY_HASH_FIELD, LOCALIZED_METADATA_HASH_FIELD, STRUCTURAL_METADATA_HASH_FIELD, "translation_model", "translation_updated"):
+            if not read_scalar(frontmatter, key):
+                skipped.append(f"missing_{key}")
```

Ruff (0.15.15)

[warning] Use `list.extend` to create a transformed list

(PERF401)